File: 001.cpp
// Sum multiples of 3 or 5 below 1000

#include <iostream>
#include <vector>

using std::vector;
using std::cout;
using std::endl;

unsigned sum_multiples_below(const vector<unsigned> multiples,
                             const unsigned limit) {
	unsigned sum = 0;
	for (unsigned i = 0; i < limit; ++i) {
		for (const auto &multiple : multiples) {
			if (i % multiple == 0) {
				sum += i;
				break;
			}
		}
	}

	return sum;
}

int main() {
	vector<unsigned> multiples {3, 5};
	unsigned limit = 1000;
	unsigned sum = sum_multiples_below(multiples, limit);
	cout << sum << endl;

	return 0;

}

File: 001.py
'''Multiples of 3 and 5'''

def is_multiple_of(n, factors):
    '''Return True if n is a multiple of any of the numbers in factors.'''

    return any(n % factor == 0 for factor in factors)

def sum_multiples_below(factors, limit):
    '''Return the sum of the multiples of any of the numbers in factors
    below limit.'''

    return sum(n for n in range(limit) if is_multiple_of(n, factors))

if __name__ == '__main__':
    print(sum_multiples_below([3, 5], 1000))
